### PR TITLE
Add a litmusbook for mongodb workload

### PR DESCRIPTION
**Pull request by shashank855**
**This PR does the following**:
- Adds a litmusbook which trigers a mongodb-loadgen job.
- The litmusbook checks for the following after deploying the loadgen:
  1. Whether the loadgen pod is in running state.
  2. Whether the loadgen has perfomed a write operation on database or not.
- Accordingly ansible task updates the result in result CR.
 
**Following is the output acheived after deploying the loadgen:**
```
PLAY [localhost] ***************************************************************

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************

TASK [Apply the litmus result CR] **********************************************

TASK [Checking the status  of test specific namespace.] ************************

TASK [Create Mongodb Loadgen Job] **********************************************
FAILED - RETRYING: Verify load-gen pod is running (15 retries left).

TASK [Verify load-gen pod is running] ******************************************

TASK [Verifying load-generation] ***********************************************

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************

TASK [Apply the litmus result CR] **********************************************

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=10   changed=8    unreachable=0    failed=0   
```
**Following is the output of load-generation:**
```
Running the test with following options:
Number of threads: 10
Report intermediate results every 10 second(s)
Initializing random number generator from current time


Initializing worker threads...

setting write concern to 1
Threads started!

[  10s] threads: 10, tps: 803.35, reads: 2412.85, writes: 2410.15, response time: 23.31ms (95%), errors: 0.00, reconnects:  0.00
[  20s] threads: 10, tps: 864.64, reads: 2594.93, writes: 2593.93, response time: 20.17ms (95%), errors: 0.00, reconnects:  0.00
```

### CODE REVIEW

File: apps/mongodb/workload/test.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+        - name: Verifying load-generation  
+          shell: kubectl exec -it mongo-0 -n {{ namespace }} -- mongo --eval 'db.getMongo().getDBNames()'
+          register: output
+          until: "'sbtest' in output.stdout"
+          delay: 30
+          retry: 15
+
+        - set_fact:
+            flag: "load_started"
```
Comment (dargasudarshan) [inline apps/mongodb/workload/test.yml]: @shashank855 Set the result flag with uniform messages like "loadgen_started", "loadgen_completed", "loadgen_failed" in all the workload litmus books.

Comment (shashank855) [reply apps/mongodb/workload/test.yml]: @dargasudarshan  sure i will make the changes

File: apps/mongodb/workload/test.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        - name: Verifying load-generation  
+          shell: kubectl exec -it mongo-0 -n {{ namespace }} -- mongo --eval 'db.getMongo().getDBNames()'
+          register: output
+          until: "'sbtest' in output.stdout"
+          delay: 30
+          retry: 15
+
+        - set_fact:
+            flag: "loadgen_started"
```
Comment (ksatchit) [inline apps/mongodb/workload/test.yml]: Let us maintain the same convention for test result flags in all litmusbooks, i.e., Pass/Fail.

File: apps/mongodb/workload/mongo_loadgen.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  template:
+    metadata:
+      name: mongo-loadgen
+      labels:
+        app: mongo-loadgen
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: mongo-loadgen
+        image: openebs/ansible-runner
```
Comment (ksatchit) [inline apps/mongodb/workload/mongo_loadgen.yml]: did you mean openebs/mongo-client here and openebs/ansible-runner in the subsequent file (run_litmus_test) ?

Comment (shashank855) [reply apps/mongodb/workload/mongo_loadgen.yml]: @ksatchit  openebs/ansible-runner will be the part of litmus job image. Where as openebs/mongo-client image will be the part of loadgen job. And this image we will crate using docker file in openebs/tet-tools

File: apps/mongodb/workload/mongo_loadgen.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: mongo-loadgen
+spec:
+  template:
+    metadata:
+      name: mongo-loadgen
+      labels:
+        app: mongo-loadgen
```
Comment (gprasath) [inline apps/mongodb/workload/mongo_loadgen.yml]: Modify the label key for loadgen pods

File: apps/mongodb/workload/mongo_loadgen.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: mongo-loadgen
+spec:
+  template:
+    metadata:
+      name: mongo-loadgen
+      labels:
+        loadgen: mongo-loadgen
```
Comment (nsathyaseelan) [inline apps/mongodb/workload/mongo_loadgen.yml]: @shashank855 can we take these label values from env

Comment (shashank855) [reply apps/mongodb/workload/mongo_loadgen.yml]: sure i will make the changes

**Pull request merged**